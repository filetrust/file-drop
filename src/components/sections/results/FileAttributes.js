import React from 'react';

function FileAttributes({ file, fileType } = {}) {
  const {name, size} = file;
  return(
    <div className="file-attributes analysis-table table-container">
      <div className="h1 table-header">File Attributes</div>
      <table>
        <tbody>
          <tr>
            <td>File Name: </td>
            <td>{name}</td>
          </tr>
          <tr>
            <td>File Size: </td>
            <td>{size}</td>
          </tr>
          <tr>
            <td>Type: </td>
            <td>{fileType}</td>
          </tr>
        </tbody>
      </table>
    </div>
  )
}

export default FileAttributes;
