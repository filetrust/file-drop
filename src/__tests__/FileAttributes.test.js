import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import FileAttributes from "../components/FileAttributes";

Enzyme.configure({adapter: new Adapter()});

const fileName = "TestFile.doc";
const fileSize = 20;
const fileType = "doc";

const mockFile = {name: fileName, size: fileSize};

describe("FileAttributes", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<FileAttributes file={mockFile} fileType={fileType}/>);
    expect(wrapper.exists()).toBe(true);
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<FileAttributes file={mockFile} fileType={fileType}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it("contains a tr for FileName", () => {
    const wrapper = shallow(<FileAttributes file={mockFile} fileType={fileType}/>);
    expect(
      wrapper.contains(
        <tr>
            <td>File Name: </td>
            <td>{mockFile.name}</td>
        </tr>
      )
    ).toBe(true);
  });

  it("contains a tr for File Size", () => {
    const wrapper = shallow(<FileAttributes file={mockFile} fileType={fileType}/>);
    expect(
      wrapper.contains(
        <tr>
            <td>File Size: </td>
            <td>{mockFile.size}</td>
        </tr>
      )
    ).toBe(true);
  });

  it("contains a tr for File Type", () => {
    const wrapper = shallow(<FileAttributes file={mockFile} fileType={fileType}/>);
    expect(
      wrapper.contains(
        <tr>
            <td>Type: </td>
            <td>{fileType}</td>
        </tr>
      )
    ).toBe(true);
  });

  it("contains a Header", () => {
    const wrapper = shallow(<FileAttributes file={mockFile} fileType={fileType}/>);
    expect(
      wrapper.contains(
        <h1 className="table-header">File Attributes</h1>
      )
    ).toBe(true);
  });
});
