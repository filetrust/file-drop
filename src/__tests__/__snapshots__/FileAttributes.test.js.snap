// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileAttributes matches snapshot 1`] = `ShallowWrapper {}`;
